tests/collections/asr/test_diar_utils.py

```diff
@@ -0,0 +1,215 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from itertools import permutations
+
+import numpy as np
+import pytest
+import torch
+
+from nemo.collections.asr.parts.utils.nmesc_clustering import SpeakerClustering
+from nemo.collections.asr.parts.utils.speaker_utils import (
+    combine_float_overlaps,
+    combine_int_overlaps,
+    get_subsegments,
+)
+
+
+def check_range_values(target, source):
+    bool_list = []
+    for tgt, src in zip(target, source):
+        for x, y in zip(src, tgt):
+            bool_list.append(abs(x - y) < 1e-6)
+    return all(bool_list)
+
+
+def matrix(mat, torch=True):
+    if torch:
+        return torch.tensor(mat)
+    else:
+        return np.array(mat)
+
+
+def generate_mock_emb(n_emb_per_spk, perturb_sigma, emb_dim):
+    """Generate a set of artificial embedding vectors from random numbers
+    """
+    return torch.rand(1, emb_dim).repeat(n_emb_per_spk, 1) + perturb_sigma * torch.rand(n_emb_per_spk, emb_dim)
+
+
+def generate_mock_data(
+    n_spks=2,
+    spk_dur=3,
+    emb_dim=192,
+    perturb_sigma=0.01,
+    ms_window=[1.5, 1.0, 0.5],
+    ms_shift=[0.75, 0.5, 0.25],
+    torch_seed=0,
+):
+
+    torch.manual_seed(torch_seed)
+    segment_lists = []
+    spk_timestamps = [(spk_dur * k, spk_dur) for k in range(n_spks)]
+    emb_list, seg_list = [], []
+    multiscale_segment_counts = [0 for _ in range(len(ms_window))]
+    for scale_idx, (window, shift) in enumerate(zip(ms_window, ms_shift)):
+        for spk_idx, (offset, dur) in enumerate(spk_timestamps):
+            segments = get_subsegments(offset=offset, window=window, shift=shift, duration=dur)
+            emb = generate_mock_emb(n_emb_per_spk=len(segments), perturb_sigma=perturb_sigma, emb_dim=emb_dim,)
+            seg_list.extend(segments)
+            emb_list.append(emb)
+            multiscale_segment_counts[scale_idx] += emb.shape[0]
+
+    emb_tensor = torch.concat(emb_list)
+    multiscale_segment_counts = torch.tensor(multiscale_segment_counts)
+    segm_tensor = torch.tensor(seg_list)
+    multiscale_weights = torch.ones(len(ms_window)).unsqueeze(0)
+    return emb_tensor, segm_tensor, multiscale_segment_counts, multiscale_weights, spk_timestamps
+
+
+@pytest.mark.skip("test is not intended")
```

Comment:
why are we skipping this?

Reply:
This function is sub-function for other testcases. This function was included in the test hogging up the time so skipped it.